Velaptor/Graphics/Renderers/TextureRenderer.cs

```diff
@@ -230,7 +315,262 @@ public void Render(
+    {
+        ArgumentNullException.ThrowIfNull(atlas);
+
+        var frames = atlas.GetFrames(subTextureName);
+
+        if (frameNumber < 0 || frameNumber >= frames.Length)
+        {
+            var exMsg =
+                $"The frame number '{frameNumber}' is invalid for atlas '{atlas.Name}' and sub-texture '{subTextureName}'." +
+                "\nThe frame number must be greater than or equal to 0 and less than or equal to the total number of frames.";
+            throw new RendererException(exMsg);
+        }
+
+        var subTextureData = frames[frameNumber];
+
+        RenderBase(
+            atlas.Texture,
+            (subTextureData.Bounds, new NETRect((int)pos.X, (int)pos.Y, (int)atlas.Width, (int)atlas.Height)),
+            0f,
+            1f,
+            color,
+            RenderEffects.None,
+            layer);
+    }
+
+    /// <inheritdoc/>
+    /// <exception cref="ArgumentNullException">Thrown if the <paramref name="atlas"/> is null.</exception>
+    /// <exception cref="InvalidOperationException">
+    ///     Thrown if the <see cref="IBatcher.Begin"/> has not been invoked before rendering.
+    /// </exception>
+    /// <exception cref="ArgumentException">
+    ///     Thrown if the source rectangle width or height is less than or equal to 0.
+    /// </exception>
+    public void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, int frameNumber = 0, int layer = 0)
+    {
+        ArgumentNullException.ThrowIfNull(atlas);
+
+        var frames = atlas.GetFrames(subTextureName);
+
+        if (frameNumber < 0 || frameNumber >= frames.Length)
+        {
+            var exMsg =
+                $"The frame number '{frameNumber}' is invalid for atlas '{atlas.Name}' and sub-texture '{subTextureName}'." +
+                "\nThe frame number must be greater than or equal to 0 and less than or equal to the total number of frames.";
+            throw new RendererException(exMsg);
+        }
+
+        var subTextureData = frames[frameNumber];
+
+        RenderBase(
+            atlas.Texture,
+            (subTextureData.Bounds, new NETRect((int)pos.X, (int)pos.Y, (int)atlas.Width, (int)atlas.Height)),
+            angle,
+            1f,
+            Color.White,
+            RenderEffects.None,
+            layer);
+    }
+
+    /// <inheritdoc/>
+    /// <exception cref="ArgumentNullException">Thrown if the <paramref name="atlas"/> is null.</exception>
+    /// <exception cref="InvalidOperationException">
+    ///     Thrown if the <see cref="IBatcher.Begin"/> has not been invoked before rendering.
+    /// </exception>
+    /// <exception cref="ArgumentException">
+    ///     Thrown if the source rectangle width or height is less than or equal to 0.
+    /// </exception>
+    public void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, int frameNumber = 0, int layer = 0)
```

Comment:
Add exception docs to all atlas render overloads that throw an exception due to an invalid frame number.